src/Features/Core/Portable/CodeRefactorings/FixAllOccurences/FixAllCodeRefactoringCodeAction.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeFixesAndRefactorings;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings
+{
+    internal sealed class FixAllCodeRefactoringCodeAction : FixAllCodeAction
+    {
+        internal FixAllCodeRefactoringCodeAction(IFixAllState fixAllState)
+            : base(fixAllState, showPreviewChangesDialog: true)
+        {
+        }
+
+        protected override IFixAllContext CreateFixAllContext(IFixAllState fixAllState, IProgressTracker progressTracker, CancellationToken cancellationToken)
+            => new FixAllContext((FixAllState)fixAllState, progressTracker, cancellationToken);
+
+        protected override bool IsInternalProvider(IFixAllState fixAllState)
```

Comment:
what is an 'internal provider'?  e.g. a first part refactoring we are shipping ourselves?

Reply:
> e.g. a first part refactoring we are shipping ourselves

Yes, that is correct. For telemetry, we log actual strings for first party fixers/refactorings, while log the hash codes for third-party ones: https://github.com/dotnet/roslyn/blob/dbedfa27696dba71d3f5cbe8a956bfa30cfacb21/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllLogger.cs#L41-L58